src/components/LoginModal/LoginModal.js

```diff
@@ -1,23 +1,54 @@
 import React, { useState } from "react";
 import PopupWithForm from "../PopupWithForm/PopupWithForm";
-const LoginModal = ({ handleModalClose, handleRegisterModal }) => {
+
+const LoginModal = ({
+  handleModalClose,
+  handleRegisterModal,
+  userSignInAccount,
+  emailNotFoundError,
+}) => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
 
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  // ERROR HANDLING
+  const validateForm = () => {
+    let isValid = true;
+
+    if (!email || !/^\S+@\S+\.\S+$/.test(email)) {
+      isValid = false;
+      setEmailError("Please enter a valid email address");
+    } else {
+      setEmailError("");
+    }
+
+    if (!password || password.length < 2 || password.length > 20) {
+      isValid = false;
+      setPasswordError("Password must be between 2 and 30 characters");
```

Comment:
If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you **don’t need** to make functions for handling inputs, storing values, resetting the form. Everything is already done.